cpp/src/fil/infer.cu

```diff
@@ -260,21 +405,35 @@ struct tree_aggregator_t<NITEMS, GROVE_PER_CLASS_FEW_CLASSES> : finalize_block {
     }
     __syncthreads();  // free up per_thread[] margin
 
-    write_best_class_in_block(to_vec(threadIdx.x, acc), num_classes, out,
-                              num_rows);
+    if (num_outputs == 1) {  // will output class
+      auto candidate = to_vec(threadIdx.x, acc);
+      write_best_class_in_block<NITEMS>(
+        &candidate, &candidate + (threadIdx.x < num_classes), tmp_storage, out,
```

Comment:
I don't think it's a good idea to use pointer arithmetic with thread's local variables.

If you want to use pointer arithmetic, consider placing the variable into shared memory.